Library/Homebrew/version/head.rb

```diff
@@ -0,0 +1,35 @@
+# typed: true
+# frozen_string_literal: true
+
+class Version
+  # A formula's HEAD version.
+  # @see https://docs.brew.sh/Formula-Cookbook#unstable-versions-head Unstable versions (head)
+  #
+  # @api private
+  class HeadVersion < Version
```

Comment:
Should this be nested like `Version::HeadVersion`? Seems weird.

Reply:
Also do we actually need this class? Seems like adding the methods here to `Version` would be simple enough.

Reply:
Seems to already be a pattern with the Version class.

https://github.com/Homebrew/brew/blob/70451ea7b14f1305bc149601e8ed74f7467c1753/Library/Homebrew/version/null.rb#L6-L10

https://github.com/Homebrew/brew/blob/70451ea7b14f1305bc149601e8ed74f7467c1753/Library/Homebrew/version/parser.rb#L4-L6

I also think refactoring that is probably out of the scope of this PR.